### PR TITLE
INT B-20083

### PR DESCRIPTION
**Pull request by brooklyn-welsh**
## Summary

This branch fixes 2 issues:
1. The Prime Simulator would not update the "Actual" weight values for Pro Gear or Spouse Pro Gear when the "Update Shipment" button was used.
2. On the "Review Weights" page during PPM Closeout, for partial shipments (HHG + PPM), the actual pro gear weights were not showing up for the HHG shipment.

To address this:
1. The prime.yaml file, updateMTOShipment service, and endpoint handler have been edited to include definitions for "actualProGear" and "actualSpouseProGear".
2. The standalone "Pro Gear" row in the Review Weights page has been removed, and the pro gear values now appear as columns in the HHG shipment's row.

### How to test

1. As a customer, create a move that contains both an HHG and a PPM shipment
2. As an SC, approve the move the usual way
3. As customer, upload PPM docs
4. As TOO, navigate to the move and approve the move for counseling.
5. As the Prime, navigate to the move, find the HHG shipment and click "Update Shipment". Here, update both "Actual" pro gear weight fields. (If you get an error saying the time to edit weights has passed, put the dates at the top of the form further into the future).
6. As the SC, navigate to the move and click "Review Shipment Weights". At the bottom of the screen, your HHG shipment should look similar to the screenshot below, but displaying whichever numbers you input as the Prime.

Screenshot 2024-05-29 at 2 45 16 PM

### CODE REVIEW

File: src/components/Customer/DodInfoForm/DodInfoForm.jsx [hunk 1]
```diff
@@ -21,18 +21,16 @@ const DodInfoForm = ({ initialValues, onSubmit, onBack, isEmplidEnabled }) => {
   // considering the edipi input when the okta_dodid_input ff is off
   const validationSchema = Yup.object().shape({
     affiliation: Yup.mixed().oneOf(Object.keys(SERVICE_MEMBER_AGENCY_LABELS)).required('Required'),
+    edipi: Yup.string()
+      .matches(/[0-9]{10}/, 'Enter a 10-digit DOD ID number')
+      .required('Required'),
     emplid: Yup.string().when('showEmplid', () => {
```
Comment (taeJungCaci) [inline src/components/Customer/DodInfoForm/DodInfoForm.jsx:27]: I was just looking at this PR to see what your resolved merge conflicts looks like and compare with mine. Saw this code stand out cause on mine I didn't see anything like this. Maybe old code? but I checked current IntegrationBranch and the Main branch and didn't find this section of code. Might be worth double checking? Not sure if it matters or not tho.

Here is my PR for ref: https://github.com/transcom/mymove/pull/12837/files#diff-3b5315b8efcc6036401065f3712c12498004eb6e37120d18839da1659d464a29

Comment (brooklyn-welsh) [reply src/components/Customer/DodInfoForm/DodInfoForm.jsx]: Good catch, seems like something that would end up causing issues, the newest push should now be identical to your PR

**Review by taeJungCaci — COMMENTED** [review]
Tested and working fine. Everything looks good except I have a question about prime api. When you update existing prime api - don't you have to update for all versions? or is it only the latest one that you update?

**Comment by brooklyn-welsh** [comment]
> Tested and working fine. Everything looks good except I have a question about prime api. When you update existing prime api - don't you have to update for all versions as long as it is not a breaking change? or is it only the latest one that you update?

That's a good question, I actually don't know, I updated v3 since it's the version the frontend was using for Prime Sim, I can ask in the dev channel and see what the response is, the Prime Versioning article they linked didn't really answer that question

**Review by JamesHawks224 — APPROVED** [review]
testing: success
code: lgtm

**Comment by brooklyn-welsh** [comment]
> > Tested and working fine. Everything looks good except I have a question about prime api. When you update existing prime api - don't you have to update for all versions as long as it is not a breaking change? or is it only the latest one that you update?
> 
> That's a good question, I actually don't know, I updated v3 since it's the version the frontend was using for Prime Sim, I can ask in the dev channel and see what the response is, the Prime Versioning article they linked didn't really answer that question

Got confirmation from Dre, you're correct, all non-breaking changes get added to all non-deprecated handlers/endpoints. I'll duplicate my changes across the other two and push up the changes

**Comment by taeJungCaci** [comment]
> > > Tested and working fine. Everything looks good except I have a question about prime api. When you update existing prime api - don't you have to update for all versions as long as it is not a breaking change? or is it only the latest one that you update?
> > 
> > 
> > That's a good question, I actually don't know, I updated v3 since it's the version the frontend was using for Prime Sim, I can ask in the dev channel and see what the response is, the Prime Versioning article they linked didn't really answer that question
> 
> Got confirmation from Dre, you're correct, all non-breaking changes get added to all non-deprecated handlers/endpoints. I'll duplicate my changes across the other two and push up the changes

Sounds good, thanks!

**Pull request merged**